File: server/routes/analytics.py
from flask import Blueprint, request, jsonify
from middleware.auth import require_auth, get_current_user_id
from services import analytics_service

analytics_bp = Blueprint('analytics', __name__, url_prefix='/api/analytics')


@analytics_bp.route('/track', methods=['POST'])
def track_event():
    data = request.get_json() or {}
    event_type = data.get('event_type')
    event_name = data.get('event_name')
    properties = data.get('properties')
    session_id = data.get('session_id')

    user_id = None
    auth_header = request.headers.get('Authorization')
    if auth_header and auth_header.startswith('Bearer '):
        user_id = get_current_user_id()

    if not event_type or not event_name:
        return jsonify({'success': False, 'message': '缺少事件参数', 'code': 400}), 400

    analytics_service.track_event(user_id, event_type, event_name, properties, session_id)
    return jsonify({'success': True, 'code': 200, 'data': {'message': '事件已记录'}})


@analytics_bp.route('/user', methods=['GET'])
@require_auth
def get_user_analytics():
    user_id = request.current_user_id
    params = request.args.to_dict()
    result = analytics_service.get_user_analytics(user_id, params)
    return jsonify(result), result.get('code', 200)


@analytics_bp.route('/ab/variant', methods=['GET'])
@require_auth
def get_user_ab_variant():
    user_id = request.current_user_id
    feature_name = request.args.get('feature_name', '')
    if not feature_name:
        return jsonify({'success': False, 'message': '缺少feature_name参数', 'code': 400}), 400
    
    result = analytics_service.get_user_ab_variant(user_id, feature_name)
    return jsonify(result), result.get('code', 200)


@analytics_bp.route('/ab/conversion', methods=['POST'])
@require_auth
def track_ab_conversion():
    user_id = request.current_user_id
    data = request.get_json() or {}
    test_id = data.get('test_id')
    variant = data.get('variant')
    conversion_type = data.get('conversion_type')

    if not test_id or not variant or not conversion_type:
        return jsonify({'success': False, 'message': '缺少必要参数', 'code': 400}), 400

    result = analytics_service.track_ab_conversion(test_id, user_id, variant, conversion_type)
    return jsonify(result), result.get('code', 200)

File: frontend/src/services/analytics.ts
import { apiClient } from './api'
import type { UserAnalytics, PlatformAnalytics, FunnelAnalysis, RetentionAnalysis, ABTest, AdminStats, ApiResponse, PaginatedResponse } from '../types'

export const analyticsService = {
  async trackEvent(data: {
    event_type: string
    event_name: string
    properties?: Record<string, any>
    session_id?: string
  }): Promise<ApiResponse<{ message: string }>> {
    const response = await apiClient.post('/analytics/track', data)
    return response.data
  },

  async getUserAnalytics(params?: {
    start_date?: string
    end_date?: string
    days?: number
  }): Promise<ApiResponse<UserAnalytics>> {
    const response = await apiClient.get('/analytics/user', { params })
    return response.data
  },

  async getABVariant(featureName: string): Promise<ApiResponse<{ variant: string; test_id?: number; test_name?: string }>> {
    const response = await apiClient.get('/analytics/ab/variant', { params: { feature_name: featureName } })
    return response.data
  },

  async trackABConversion(data: {
    test_id: number
    variant: string
    conversion_type: string
  }): Promise<ApiResponse<{ message: string }>> {
    const response = await apiClient.post('/analytics/ab/conversion', data)
    return response.data
  },

  async getPlatformAnalytics(params?: { days?: number }): Promise<ApiResponse<PlatformAnalytics>> {
    const response = await apiClient.get('/admin/analytics/platform', { params })
    return response.data
  },

  async getFunnelAnalysis(params?: { funnel?: string; days?: number }): Promise<ApiResponse<FunnelAnalysis>> {
    const response = await apiClient.get('/admin/analytics/funnel', { params })
    return response.data
  },

  async getRetentionAnalysis(params?: { days?: number; cohort_days?: number }): Promise<ApiResponse<RetentionAnalysis>> {
    const response = await apiClient.get('/admin/analytics/retention', { params })
    return response.data
  },

  async getAdminStats(): Promise<ApiResponse<AdminStats>> {
    const response = await apiClient.get('/admin/stats')
    return response.data
  },

  async getABTests(params?: { is_active?: boolean; page?: number; per_page?: number }): Promise<ApiResponse<PaginatedResponse<ABTest>>> {
    const response = await apiClient.get('/admin/ab-tests', { params })
    return response.data
  },

  async createABTest(data: {
    name: string
    feature_name: string
    variants: string[]
    traffic_split: Record<string, number>
  }): Promise<ApiResponse<ABTest>> {
    const response = await apiClient.post('/admin/ab-tests', data)
    return response.data
  }
}
